Support 32bit architecture for keyring Operations

Building on i686 failed with the following error:
```
BUILDSTDERR:      Running `/usr/bin/rustc --crate-name libstratis --edition=2018 src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -Cembed-bitcode=no --cfg 'feature="dbus"' --cfg 'feature="dbus_enabled"' --cfg 'feature="default"' --cfg 'feature="libdbus-sys"' --cfg 'feature="regex"' -C metadata=81f99a3984f79075 -C extra-filename=-81f99a3984f79075 --out-dir /builddir/build/BUILD/stratisd-2.2.0/target/release/deps -L dependency=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps --extern byteorder=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libbyteorder-b12384a4243db718.rmeta --extern chrono=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libchrono-13e8a1890aaf0993.rmeta --extern clap=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libclap-660ea15b71c47ce4.rmeta --extern crc=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libcrc-d2e3bcf57be3f6e1.rmeta --extern dbus=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libdbus-f4b613a1c936ee22.rmeta --extern devicemapper=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libdevicemapper-4c5214aff4b47167.rmeta --extern env_logger=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libenv_logger-197ced80a5139a02.rmeta --extern itertools=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libitertools-f08f9d68056feb20.rmeta --extern lazy_static=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/liblazy_static-92f8f8028aed6411.rmeta --extern libc=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/liblibc-64f0a9878807dca9.rmeta --extern libcryptsetup_rs=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/liblibcryptsetup_rs-96a951aba64af548.rmeta --extern libdbus_sys=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/liblibdbus_sys-c509411798aa945e.rmeta --extern libmount=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/liblibmount-12774d2a64830408.rmeta --extern libudev=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/liblibudev-46cf559e81c3931e.rmeta --extern log=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/liblog-229fb1eeaf681f47.rmeta --extern nix=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libnix-c3d1e34f845da9cf.rmeta --extern rand=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/librand-4c7f45343728c4f1.rmeta --extern regex=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libregex-f32abda88c14ca77.rmeta --extern semver=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libsemver-1583bfc57f78d4c8.rmeta --extern serde=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libserde-f701841279ce53c5.rmeta --extern serde_derive=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libserde_derive-31a30ff7e2fed296.so --extern serde_json=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libserde_json-675a91d1e290dc79.rmeta --extern tempfile=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libtempfile-414cf892603f4f19.rmeta --extern termios=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libtermios-7996172d070c6006.rmeta --extern timerfd=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libtimerfd-6e5af0f08d3afbe1.rmeta --extern uuid=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libuuid-5a1438b971d20e14.rmeta -Copt-level=3 -Cdebuginfo=2 -Clink-arg=-Wl,-z,relro,-z,now -L native=/usr/lib -L native=/usr/lib -l cryptsetup -l blkid -L native=/usr/lib -L native=/builddir/build/BUILD/stratisd-2.2.0/target/release/build/backtrace-sys-117d7a6f0ba5b992/out/.libs -L native=/usr/lib -L native=/usr/lib -L native=/usr/lib`
BUILDSTDERR: error[E0308]: mismatched types
BUILDSTDERR:    --> src/engine/macros.rs:196:67
BUILDSTDERR:     |
BUILDSTDERR: 196 |         <$to_type as std::convert::TryFrom<$from_type>>::try_from(expr).map_err(|_| {
BUILDSTDERR:     |                                                                   ^^^^
BUILDSTDERR:     |                                                                   |
BUILDSTDERR:     |                                                                   expected `i64`, found `i32`
BUILDSTDERR:     |                                                                   help: you can convert an `i32` to `i64`: `expr.into()`
BUILDSTDERR:     | 
BUILDSTDERR:    ::: src/engine/strat_engine/keys.rs:52:14
BUILDSTDERR:     |
BUILDSTDERR: 52  |         i => convert_int!(i, i64, KeySerial),
BUILDSTDERR:     |              ------------------------------- in this macro invocation
BUILDSTDERR:     |
BUILDSTDERR:     = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
BUILDSTDERR: error[E0308]: mismatched types
BUILDSTDERR:    --> src/engine/macros.rs:196:67
BUILDSTDERR:     |
BUILDSTDERR: 196 |         <$to_type as std::convert::TryFrom<$from_type>>::try_from(expr).map_err(|_| {
BUILDSTDERR:     |                                                                   ^^^^
BUILDSTDERR:     |                                                                   |
BUILDSTDERR:     |                                                                   expected `i64`, found `i32`
BUILDSTDERR:     |                                                                   help: you can convert an `i32` to `i64`: `expr.into()`
BUILDSTDERR:     | 
BUILDSTDERR:    ::: src/engine/strat_engine/keys.rs:86:9
BUILDSTDERR:     |
BUILDSTDERR: 86  |         convert_int!(key_id, i64, KeySerial).map(Some)
BUILDSTDERR:     |         ------------------------------------ in this macro invocation
BUILDSTDERR:     |
BUILDSTDERR:     = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
BUILDSTDERR: error[E0308]: mismatched types
BUILDSTDERR:    --> src/engine/macros.rs:196:67
BUILDSTDERR:     |
BUILDSTDERR: 196 |         <$to_type as std::convert::TryFrom<$from_type>>::try_from(expr).map_err(|_| {
BUILDSTDERR:     |                                                                   ^^^^
BUILDSTDERR:     |                                                                   |
BUILDSTDERR:     |                                                                   expected `i64`, found `i32`
BUILDSTDERR:     |                                                                   help: you can convert an `i32` to `i64`: `expr.into()`
BUILDSTDERR:     | 
BUILDSTDERR:    ::: src/engine/strat_engine/keys.rs:124:45
BUILDSTDERR:     |
BUILDSTDERR: 124 |             SizedKeyMemory::new(key_buffer, convert_int!(i, i64, usize)?),
BUILDSTDERR:     |                                             --------------------------- in this macro invocation
BUILDSTDERR:     |
BUILDSTDERR:     = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
BUILDSTDERR: error[E0308]: mismatched types
BUILDSTDERR:    --> src/engine/macros.rs:196:67
BUILDSTDERR:     |
BUILDSTDERR: 196 |         <$to_type as std::convert::TryFrom<$from_type>>::try_from(expr).map_err(|_| {
BUILDSTDERR:     |                                                                   ^^^^
BUILDSTDERR:     |                                                                   |
BUILDSTDERR:     |                                                                   expected `i64`, found `i32`
BUILDSTDERR:     |                                                                   help: you can convert an `i32` to `i64`: `expr.into()`
BUILDSTDERR:     | 
BUILDSTDERR:    ::: src/engine/strat_engine/keys.rs:268:22
BUILDSTDERR:     |
BUILDSTDERR: 268 |                 i => convert_int!(i, i64, usize)?,
BUILDSTDERR:     |                      --------------------------- in this macro invocation
BUILDSTDERR:     |
BUILDSTDERR:     = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
BUILDSTDERR: error[E0308]: mismatched types
BUILDSTDERR:    --> src/engine/macros.rs:196:67
BUILDSTDERR:     |
BUILDSTDERR: 196 |         <$to_type as std::convert::TryFrom<$from_type>>::try_from(expr).map_err(|_| {
BUILDSTDERR:     |                                                                   ^^^^
BUILDSTDERR:     |                                                                   |
BUILDSTDERR:     |                                                                   expected `i64`, found `i32`
BUILDSTDERR:     |                                                                   help: you can convert an `i32` to `i64`: `expr.into()`
BUILDSTDERR:     | 
BUILDSTDERR:    ::: src/engine/strat_engine/keys.rs:307:26
BUILDSTDERR:     |
BUILDSTDERR: 307 |                     i => convert_int!(i, i64, usize)?,
BUILDSTDERR:     |                          --------------------------- in this macro invocation
BUILDSTDERR:     |
BUILDSTDERR:     = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
BUILDSTDERR: error: aborting due to 5 previous errors
BUILDSTDERR: For more information about this error, try `rustc --explain E0308`.
BUILDSTDERR: error: could not compile `libstratis`.
BUILDSTDERR: Caused by:
BUILDSTDERR:   process didn't exit successfully: `/usr/bin/rustc --crate-name libstratis --edition=2018 src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -Cembed-bitcode=no --cfg 'feature="dbus"' --cfg 'feature="dbus_enabled"' --cfg 'feature="default"' --cfg 'feature="libdbus-sys"' --cfg 'feature="regex"' -C metadata=81f99a3984f79075 -C extra-filename=-81f99a3984f79075 --out-dir /builddir/build/BUILD/stratisd-2.2.0/target/release/deps -L dependency=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps --extern byteorder=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libbyteorder-b12384a4243db718.rmeta --extern chrono=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libchrono-13e8a1890aaf0993.rmeta --extern clap=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libclap-660ea15b71c47ce4.rmeta --extern crc=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libcrc-d2e3bcf57be3f6e1.rmeta --extern dbus=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libdbus-f4b613a1c936ee22.rmeta --extern devicemapper=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libdevicemapper-4c5214aff4b47167.rmeta --extern env_logger=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libenv_logger-197ced80a5139a02.rmeta --extern itertools=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libitertools-f08f9d68056feb20.rmeta --extern lazy_static=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/liblazy_static-92f8f8028aed6411.rmeta --extern libc=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/liblibc-64f0a9878807dca9.rmeta --extern libcryptsetup_rs=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/liblibcryptsetup_rs-96a951aba64af548.rmeta --extern libdbus_sys=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/liblibdbus_sys-c509411798aa945e.rmeta --extern libmount=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/liblibmount-12774d2a64830408.rmeta --extern libudev=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/liblibudev-46cf559e81c3931e.rmeta --extern log=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/liblog-229fb1eeaf681f47.rmeta --extern nix=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libnix-c3d1e34f845da9cf.rmeta --extern rand=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/librand-4c7f45343728c4f1.rmeta --extern regex=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libregex-f32abda88c14ca77.rmeta --extern semver=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libsemver-1583bfc57f78d4c8.rmeta --extern serde=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libserde-f701841279ce53c5.rmeta --extern serde_derive=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libserde_derive-31a30ff7e2fed296.so --extern serde_json=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libserde_json-675a91d1e290dc79.rmeta --extern tempfile=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libtempfile-414cf892603f4f19.rmeta --extern termios=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libtermios-7996172d070c6006.rmeta --extern timerfd=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libtimerfd-6e5af0f08d3afbe1.rmeta --extern uuid=/builddir/build/BUILD/stratisd-2.2.0/target/release/deps/libuuid-5a1438b971d20e14.rmeta -Copt-level=3 -Cdebuginfo=2 -Clink-arg=-Wl,-z,relro,-z,now -L native=/usr/lib -L native=/usr/lib -l cryptsetup -l blkid -L native=/usr/lib -L native=/builddir/build/BUILD/stratisd-2.2.0/target/release/build/backtrace-sys-117d7a6f0ba5b992/out/.libs -L native=/usr/lib -L native=/usr/lib -L native=/usr/lib` (exit code: 1)
BUILDSTDERR: error: Bad exit status from /var/tmp/rpm-tmp.xb6EHz (%build)
RPM build errors:
BUILDSTDERR:     Bad exit status from /var/tmp/rpm-tmp.xb6EHz (%build)
Child return code was: 1
EXCEPTION: [Error()]
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/mockbuild/trace_decorator.py", line 96, in trace
    result = func(*args, **kw)
  File "/usr/lib/python3.6/site-packages/mockbuild/util.py", line 700, in do
    raise exception.Error("Command failed: \n # %s\n%s" % (command, output), child.returncode)
mockbuild.exception.Error: Command failed: 
 # bash --login -c /usr/bin/rpmbuild -bb --target i686 --nodeps /builddir/build/SPECS/stratisd.spec
```

--- a/src/engine/strat_engine/backstore/crypt.rs
+++ b/src/engine/strat_engine/backstore/crypt.rs
@@ -1063,7 +1063,8 @@ mod tests {
             if fstat_result < 0 {
                 return Err(Box::new(io::Error::last_os_error()));
             }
-            let device_size = convert_int!(unsafe { stat.assume_init() }.st_size, i64, usize)?;
+            let device_size =
+                convert_int!(unsafe { stat.assume_init() }.st_size, libc::off_t, usize)?;
             let mapped_ptr = unsafe {
                 libc::mmap(
                     ptr::null_mut(),

--- a/src/engine/strat_engine/keys.rs
+++ b/src/engine/strat_engine/keys.rs
@@ -49,7 +49,7 @@ fn get_persistent_keyring() -> StratisResult<KeySerial> {
         )
     } {
         i if i < 0 => Err(io::Error::last_os_error().into()),
-        i => convert_int!(i, i64, KeySerial),
+        i => convert_int!(i, libc::c_long, KeySerial),
     }
 }
 
@@ -83,7 +83,7 @@ fn search_key(
             Err(io::Error::last_os_error().into())
         }
     } else {
-        convert_int!(key_id, i64, KeySerial).map(Some)
+        convert_int!(key_id, libc::c_long, KeySerial).map(Some)
     }
 }
 
@@ -121,7 +121,7 @@ fn read_key(
         i if i < 0 => Err(io::Error::last_os_error().into()),
         i => Ok(Some((
             key_id as KeySerial,
-            SizedKeyMemory::new(key_buffer, convert_int!(i, i64, usize)?),
+            SizedKeyMemory::new(key_buffer, convert_int!(i, libc::c_long, usize)?),
         ))),
     }
 }
@@ -265,7 +265,7 @@ impl KeyIdList {
                 )
             } {
                 i if i < 0 => return Err(io::Error::last_os_error().into()),
-                i => convert_int!(i, i64, usize)?,
+                i => convert_int!(i, libc::c_long, usize)?,
             };
 
             let num_key_ids = num_bytes_read / size_of::<KeySerial>();
@@ -304,7 +304,7 @@ impl KeyIdList {
                     )
                 } {
                     i if i < 0 => return Err(io::Error::last_os_error().into()),
-                    i => convert_int!(i, i64, usize)?,
+                    i => convert_int!(i, libc::c_long, usize)?,
                 };
 
                 if len <= keyctl_buffer.capacity() {

